pkg/encrypt/encrypt.go

```diff
@@ -0,0 +1,72 @@
+package encrypt
+
+import (
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"golang.org/x/crypto/chacha20poly1305"
+)
+
+const prefix = "ENC*"
+
+type Cipher interface {
+	Decrypt(string) (string, error)
+	Encrypt(string) (string, error)
+}
+
+var _ Cipher = (*ChaChaCipher)(nil)
+
+type ChaChaCipher struct {
```

Comment:
given this contains an AEAD interface, maybe calling it `type cipher struct` would be fine - or aeadCipher or something